app/graphql/types/query.rb

```diff
@@ -6,6 +6,20 @@ class Query < Types::BaseObject
     graphql_name 'Query'
     description 'The root of all queries'
 
+    class << self
+      def record_field(name, type)
+        field name, type, resolver: Resolvers::Generic.for(type).record
+      end
+
+      def collection_field(name, type)
+        field name, type.connection_type, null: false, resolver: Resolvers::Generic.for(type).collection
+      end
+    end
+
+    collection_field :systems, Types::System
+    collection_field :profiles, Types::Profile
+    record_field :profile, Types::Profile
+
     field :all_systems, [Types::System], null: true do
```

Comment:
Shouldn't we remove this? Or is this for backwards compatibility?

Reply:
Yeah, I would like to keep these until we have this in prod so we can phase out the old API when the new one is already used everywhere.